presto-router/src/main/java/com/facebook/presto/router/cluster/ClusterManager.java

```diff
@@ -105,4 +218,126 @@ private void initializeServerWeights()
             }
         });
     }
+
+    private void initializeMembersDiscoveryURI()
+    {
+        groups.forEach((name, groupSpec) -> {
+            List<URI> members = groupSpec.getMembers();
+            List<URI> membersDiscoveryURI = groupSpec.getMembersDiscoveryURI();
+            for (int i = 0; i < members.size(); i++) {
+                discoveryURIs.put(members.get(i), membersDiscoveryURI.get(i));
+            }
+        });
+    }
+
+    public void refreshHealthStatuses()
+    {
+        List<RemoteState> remoteStates = new ArrayList<>(remoteClusterInfos.values());
+        remoteStates.addAll(remoteQueryInfos.values());
+        remoteStates.forEach(RemoteState::asyncRefresh);
+    }
+
+    public ConcurrentHashMap<URI, RemoteClusterInfo> getRemoteClusterInfos()
```

Comment:
`getRemoteClusterInfos` and `getRemoteQueryInfos` can be made internal now (used only withing this class and `TestHealthChecks`). Also annotate these with `@VisibleForTesting`